src/loadNamespaces.tsx

```diff
@@ -43,12 +44,22 @@ export default async function loadNamespaces(
   const defaultLoader: LocaleLoader = () => Promise.resolve({})
   const pageNamespaces =
     (await Promise.all(
-      namespaces.map((ns) =>
-        typeof conf.loadLocaleFrom === 'function'
-          ? conf.loadLocaleFrom(__lang, ns)
-          : defaultLoader(__lang, ns)
+      namespaces.map(
+        (ns) =>
+          new Promise<I18nDictionary>((resolve) => {
```

Comment:
It's not necessary to create a new `Promise`. It return already a promise.

I would change it for just:

```js
typeof conf.loadLocaleFrom === 'function'
          ? conf.loadLocaleFrom(__lang, ns).catch(() => ({}))
          : defaultLoader(__lang, ns)
```